eng/mgmt/automation/sdk_generate.py

```diff
@@ -0,0 +1,264 @@
+            'packageFolder':
+                output_folder,
+        })
+
+        update_parameters(pre_suffix)
+
+    if not packages:
+        # try data-plane codegen
+        packages = sdk_automation_data(config)
+
+    with open(output_file, 'w') as fout:
+        output = {
+            'packages': packages,
+        }
+        json.dump(output, fout)
+
+
+def main():
+    args = vars(parse_args())
+
+    if args.get('config'):
+        return sdk_automation(args['config'][0], args['config'][1])
+
+    base_dir = os.path.abspath(os.path.dirname(sys.argv[0]))
+    sdk_root = os.path.abspath(os.path.join(base_dir, SDK_ROOT))
+    api_specs_file = os.path.join(base_dir, API_SPECS_FILE)
+
+    readme = args['readme']
+    match = re.match(
+        'specification/([^/]+)/resource-manager/readme.md',
+        readme,
+        re.IGNORECASE,
+    )
+    if not match:
+        spec = readme
+        readme = 'specification/{0}/resource-manager/readme.md'.format(spec)
+    else:
+        spec = match.group(1)
+
+    args['readme'] = readme
+    args['spec'] = spec
+
+    # update_parameters(
+    #     args.get('suffix') or get_suffix_from_api_specs(api_specs_file, spec))
+    update_parameters(args.get('suffix'))
+    service = get_and_update_service_from_api_specs(api_specs_file, spec,
+                                                    args['service'])
+    args['service'] = service
+    module = ARTIFACT_FORMAT.format(service)
+    stable_version, current_version = set_or_increase_version(sdk_root, GROUP_ID, module, **args)
+    args['version'] = current_version
+    output_folder = OUTPUT_FOLDER_FORMAT.format(service),
+    namespace = NAMESPACE_FORMAT.format(service)
+    succeeded = generate(
+        sdk_root,
+        module = module,
+        output_folder = output_folder,
+        namespace = namespace,
+        **args
+    )
+
+    if succeeded:
+        succeeded = compile_package(sdk_root, module)
+        if succeeded:
+            compare_with_maven_package(sdk_root, service, stable_version,
+                                    current_version, module)
+
+            if args.get('auto_commit_external_change') and args.get(
+                    'user_name') and args.get('user_email'):
+                pwd = os.getcwd()
+                try:
+                    os.chdir(sdk_root)
+                    os.system('git add eng/versioning eng/mgmt pom.xml {0} {1}'.format(
+                        CI_FILE_FORMAT.format(service),
+                        POM_FILE_FORMAT.format(service)))
+                    os.system(
+                        'git -c user.name={0} -c user.email={1} commit -m "[Automation] External Change"'
+                        .format(args['user_name'], args['user_email']))
+                finally:
+                    os.chdir(pwd)
+
+    if not succeeded:
+        raise RuntimeError('Failed to generate code or compile the package')
+
+
+if __name__ == '__main__':
+    logging.basicConfig(
```

Comment:
nit, refresh main, consider use `stream=sys.stdout,` here (if you do not have reason to log to stderr).

Reply:
update the file with  your changes is generate.py, thx